src/dvclive/live.py

```diff
@@ -155,7 +180,20 @@ def set_step(self, step: int) -> None:
                 data.dump(data.val, self._step)
             self.make_summary()
 
-        self.make_report()
+        if self._report == "studio":
+            response = post_to_studio(self, "data")
+            logger.debug("Post to Studio `data` event")
+            if not response or response.status_code != 200:
+                logger.warning(
+                    "`post_to_studio` `data` event failed."
+                    "Data will be resent on next call."
+                )
+                if response:
+                    logger.debug(response.content.decode())
+            else:
+                self._latest_studio_step = step
```

Comment:
```suggestion
        if self._report == "studio":
            response = post_to_studio(self, "data")
            logger.debug("Post to Studio `data` event")
            if response:
                self._latest_studio_step = step
            else:   
                logger.warning(
                    "`post_to_studio` `data` event failed."
                    "Data will be resent on next call."
                )
                if message := response.content.decode():
                    logger.debug(message)
```

Reply:
Although the conditional for `logger.debug(message)` might be a bit odd in some cases, might be worth

```python
message = response.content.decode()
logger.debug(f"post_to_studio: {response.status_code=} " + f", {message=}" if message else "")
```

Reply:
Moved the `debug` messages to inside `post_to_studio` and left only the `warning` in `Live` class, please TAL